cmd/janus/main.go

```diff
@@ -65,7 +65,7 @@ func main() {
 	defer accessor.Close()
 	defer statsdClient.Close()
 
-	statsClient := stats.NewStatsClient(statsdClient)
+	statsClient := stats.NewStatsClient(statsdClient, globalConfig.Statsd.DSN == "")
```

Comment:
I understand the motivation of doing this and I agree that we should have it, just the implementation that is not really nice. If we create an interface for the `StatsClient` we could have the real `StatsClient` and if `globalConfig.Statsd.DSN == ""` we would use the `MutedClient` that only logs (strategy pattern), this way we wouldn't need to change the contract to have mute option that makes it not really clear.